qiskit/algorithms/evolvers/variational/variational_principles/imaginary_mc_lachlan_variational_principle.py

```diff
@@ -0,0 +1,81 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class for an Imaginary McLachlan's Variational Principle."""
+
+from typing import Union, List
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.opflow import StateFn, OperatorBase, ListOp
+from ..calculators import (
+    metric_tensor_calculator,
+    evolution_grad_calculator,
+)
+from .imaginary_variational_principle import (
+    ImaginaryVariationalPrinciple,
+)
+
+
+class ImaginaryMcLachlanVariationalPrinciple(ImaginaryVariationalPrinciple):
```

Comment:
What do you think about dropping `VariationalPrinciple` from the name for brevity, so it's just `ImaginaryMcLachlan`? In the context of variational time evolution it's probably clear that it's a variational principle 🤔